document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreQueryV1Test.java

```diff
@@ -3674,4 +3749,391 @@ private static void testCountApi(
     final long expectedSize = convertJsonToMap(fileContent).size();
     assertEquals(expectedSize, actualSize);
   }
+
+  @ParameterizedTest
+  @ArgumentsSource(PostgresProvider.class)
+  void testFlatPostgresCollectionFindAll(String dataStoreName) throws IOException {
```

Comment:
If all these tests are specific to Postgres, worth nesting it under a nested class with `@Nested` annotation. That way, all related tests would be grouped together.

Also, it would be better to move all the tests (classes and methods) to the beginning of the class (before all the private methods).